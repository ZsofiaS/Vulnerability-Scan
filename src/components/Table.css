.table {
  table-layout: fixed;
  width: 100%;
  border-collapse: collapse;
  text-align: center;
}
th, td {
  padding: 0.5rem 0.3rem;
}
.header {
  background-color: #F4F6F6;
}
