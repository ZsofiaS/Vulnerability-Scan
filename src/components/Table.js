import React from 'react';
import data from '../data/scan';
import TableRow from './TableRow';

const Table = () => {

  const renderTable = () => {
    return data.scan.vulnerabilities.map((vulnerability, i) => {
      const { id, severity, name, description, solution, references, cvssBaseScore, affectedAssets } = vulnerability;
      return (
        <TableRow
          key={i}
          severity={severity}
          name={name}
          description={description}
          solution={solution}
          references={references}
          cvssBaseScore={cvssBaseScore}
          affectedAssets={affectedAssets}
          renderAssets={renderAssets(affectedAssets)}
        />
      )
    })
  }

  const renderAssets = (assets) => {
    return assets.map((asset, i) => {
      let item = data.scan.assets.find(obj => obj.id == asset);
      return (
        <span key={i}>{item.description} </span>
      )
    })
  }

  return (
    <table>
      <thead>
        <tr>
          <th>Severity</th>
          <th>Issue</th>
          {/*
          <th>Summary</th>
          <th>Solution</th>
          <th>References</th>
          */}
          <th>CVSS Base Score</th>
          <th>Affected Assets</th>
        </tr>
      </thead>
      <tbody>
        {renderTable()}
      </tbody>
    </table>
  )
}

export default Table;
