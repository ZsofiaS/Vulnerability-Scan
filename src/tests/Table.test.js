import React from 'react';
import { shallow } from 'enzyme';
import Table from '../components/Table';
import toJson from 'enzyme-to-json';

describe('renders Table component', () => {
  it('renders without crashing', () => {
    shallow(<Table />);
  });
  it('renders table header', () => {
    const wrapper = shallow(<Table />);
    const tableHeader = (<th>Severity</th>);
    expect(wrapper.contains(tableHeader)).toEqual(true);
  });
});

describe('snapshot', () => {
  it('Table snapshot', () => {
    const tree = shallow(<Table />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});
