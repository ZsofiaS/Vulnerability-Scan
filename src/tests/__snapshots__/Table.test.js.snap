// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot Table snapshot 1`] = `
<table>
  <thead>
    <tr>
      <th>
        Severity
      </th>
      <th>
        Issue
      </th>
      <th>
        CVSS Base Score
      </th>
      <th>
        Affected Assets
      </th>
      <th>
        More info
      </th>
    </tr>
  </thead>
  <tbody>
    <TableRow
      affectedAssets={
        Array [
          33,
        ]
      }
      cvssBaseScore="7.5"
      description="ext/standard/var_unserializer.re in PHP 7.0.x through 7.0.22 and 7.1.x through 7.1.8 is prone to a heap use after free while unserializing untrusted data, related to improper use of the hash API for key deletion in a situation with an invalid array size. Exploitation of this issue can have an unspecified impact on the integrity of PHP."
      key="0"
      name="PHP Vulnerability: CVE-2017-12932"
      references="https://bugs.php.net/bug.php?id=74103 "
      renderAssets={
        Array [
          <p>
            Main Database
             
          </p>,
        ]
      }
      severity="high"
      solution="Download and apply the upgrade from: Download and apply the upgrade from:"
    />
    <TableRow
      affectedAssets={
        Array [
          9,
        ]
      }
      cvssBaseScore="7.5"
      description="In PHP through 5.6.33, 7.0.x before 7.0.28, 7.1.x through 7.1.14, and 7.2.x through 7.2.2, there is a stack-based buffer under-read while parsing an HTTP response in the php_stream_url_wrap_http_ex function in ext/standard/http_fopen_wrapper.c. This subsequently results in copying a large string."
      key="1"
      name="PHP Vulnerability: CVE-2018-7584"
      references={null}
      renderAssets={
        Array [
          <p>
            Email server
             
          </p>,
        ]
      }
      severity="high"
      solution="Download and apply the upgrade from: Download and apply the upgrade from: Download and apply the upgrade from: Download and apply the upgrade from:"
    />
    <TableRow
      affectedAssets={
        Array [
          9,
          33,
        ]
      }
      cvssBaseScore="5.0"
      description="The PHP install on the remote server is configured in a way that allows disclosure of potentially sensitive information to an attacker through a special URL. Such a URL triggers an Easter egg built into PHP itself. Other such Easter eggs likely exist, but Nessus has not checked for them."
      key="2"
      name="PHP expose_php Information Disclosure"
      references="http://www.0php.com/php_easter_egg.php http://seclists.org/webappsec/2004/q4/324"
      renderAssets={
        Array [
          <p>
            Email server
             
          </p>,
          <p>
            Main Database
             
          </p>,
        ]
      }
      severity="medium"
      solution="In the PHP configuration file, php.ini, set the value for 'expose_php' to 'Off' to disable this behavior. Restart the web server daemon to put this change into effect."
    />
    <TableRow
      affectedAssets={
        Array [
          9,
          33,
          35,
        ]
      }
      cvssBaseScore="5.0"
      description="The remote web server supports the TRACE and/or TRACK methods. TRACE and TRACK are HTTP methods that are used to debug web server connections."
      key="3"
      name="HTTP TRACE / TRACK Methods Allowed"
      references="http://www.cgisecurity.com/whitehat-mirror/WH-WhitePaper_XST_ebook.pdf http://www.apacheweek.com/issues/03-01-24 http://download.oracle.com/sunalerts/1000718.1.html"
      renderAssets={
        Array [
          <p>
            Email server
             
          </p>,
          <p>
            Main Database
             
          </p>,
          <p>
            Firewall
             
          </p>,
        ]
      }
      severity="medium"
      solution="Disable these methods. Refer to the plugin output for more information."
    />
    <TableRow
      affectedAssets={
        Array [
          9,
          33,
        ]
      }
      cvssBaseScore="2.6"
      description="The server is configured to support ciphers known as static key ciphers. These ciphers don't support 'Forward Secrecy'. In the new specification for HTTP/2, these ciphers have been blacklisted."
      key="4"
      name="TLS/SSL Server Supports The Use of Static Key Ciphers"
      references="http://www.nist.gov/manuscript-publication-search.cfm?pub_id=915295 https://wiki.mozilla.org/Security/Server_Side_TLS https://www.owasp.org/index.php/Transport_Layer_Protection_Cheat_Sheet#Rule_-_Only_Support_Strong_Cryptographic_Ciphers http://support.microsoft.com/kb/245030/ https://tools.ietf.org/html/rfc7540/"
      renderAssets={
        Array [
          <p>
            Email server
             
          </p>,
          <p>
            Main Database
             
          </p>,
        ]
      }
      severity="low"
      solution="Configure the server to disable support for static key cipher suites. For Microsoft IIS web servers, see Microsoft Knowledgebase article http://support.microsoft.com/kb/245030/ (245030) for instructions on disabling static key cipher suites. The following recommended configuration provides a higher level of security. This configuration is compatible with Firefox 27, Chrome 22, IE 11, Opera 14 and Safari 7. SSLv2, SSLv3, and TLSv1 protocols are not recommended in this configuration. Instead, use TLSv1.1 and TLSv1.2 protocols. Refer to your server vendor documentation to apply the recommended cipher configuration: ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-AES256-GCM-SHA384:DHE-RSA-AES128-GCM-SHA256:DHE-DSS-AES128-GCM-SHA256:kEDH+AESGCM:ECDHE-RSA-AES128-SHA256:ECDHE-ECDSA-AES128-SHA256:ECDHE-RSA-AES128-SHA:ECDHE-ECDSA-AES128-SHA:ECDHE-RSA-AES256-SHA384:ECDHE-ECDSA-AES256-SHA384:ECDHE-RSA-AES256-SHA:ECDHE-ECDSA-AES256-SHA:DHE-RSA-AES128-SHA256:DHE-RSA-AES128-SHA:DHE-DSS-AES128-SHA256:DHE-RSA-AES256-SHA256:DHE-DSS-AES256-SHA:DHE-RSA-AES256-SHA:!aNULL:!eNULL:!EXPORT:!DES:!RC4:!3DES:!MD5:!PSK"
    />
    <TableRow
      affectedAssets={
        Array [
          9,
          33,
          35,
        ]
      }
      cvssBaseScore="0.0"
      description="By calling the OPTIONS method, it is possible to determine which HTTP methods are allowed on each directory. As this list may be incomplete, the plugin also tests - if 'Thorough tests' are enabled or 'Enable web applications tests' is set to 'yes' in the scan policy - various known HTTP methods on each directory and considers them as unsupported if it receives a response code of 400, 403, 405, or 501. Note that the plugin output is only informational and does not necessarily indicate the presence of any security vulnerabilities."
      key="5"
      name="HTTP Methods Allowed (per directory)"
      references={null}
      renderAssets={
        Array [
          <p>
            Email server
             
          </p>,
          <p>
            Main Database
             
          </p>,
          <p>
            Firewall
             
          </p>,
        ]
      }
      severity="information"
      solution={null}
    />
    <TableRow
      affectedAssets={
        Array [
          9,
          33,
          35,
        ]
      }
      cvssBaseScore="0.0"
      description="This plugin is a SYN 'half-open' port scanner. It shall be reasonably quick even against a firewalled target. Note that SYN scans are less intrusive than TCP (full connect) scans against broken services, but they might cause problems for less robust firewalls and also leave unclosed connections on the remote target, if the network is loaded."
      key="6"
      name="Nessus SYN scanner"
      references={null}
      renderAssets={
        Array [
          <p>
            Email server
             
          </p>,
          <p>
            Main Database
             
          </p>,
          <p>
            Firewall
             
          </p>,
        ]
      }
      severity="information"
      solution="Protect your target with an IP filter."
    />
  </tbody>
</table>
`;
